docs/intro/version_history.rst

```diff
@@ -1,6 +1,48 @@
 Version history
 ---------------
 
+1.4.0
+=====
+
+* OSX is :repo:`now supported </bazel#quick-start-bazel-build-for-developers>`. (A few features are
+  missing such as hot restart and original destination routing).
+* Log verbosity :repo:`compile time flag </bazel#log-verbosity>` added.
+* Hot restart :repo:`compile time flag </bazel#hot-restart>` added.
+* Original destination :ref:`cluster <arch_overview_service_discovery_types_original_destination>`
+  and :ref:`load balancer <arch_overview_load_balancing_types_original_destination>` added.
+* :ref:`WebSocket <arch_overview_websocket>` is now supported.
+* Virtual cluster priorities have been hard removed without deprecation as we are reasonably sure
+  no one is using this feature.
+* Route :ref:`validate_clusters <config_http_conn_man_route_table_validate_clusters>` option added.
+* :ref:`x-envoy-downstream-service-node <config_http_conn_man_headers_downstream-service-node>`
+  header added.
+* :ref:`x-forwarded-client-cert <config_http_conn_man_headers_x-forwarded-client-cert>` header
+  added.
+* Initial HTTP/1 forward proxy support for :ref:`absolute URLs
+  <config_http_conn_man_http1_settings>` has been added.
+* HTTP/2 codec settings are now :ref:`configurable <config_http_conn_man_http2_settings>`.
+* gRPC/JSON transcoder :ref:`filter <config_http_filters_grpc_json_transcoder>` added.
+* gRPC web :ref:`filter <config_http_filters_grpc_web>` added.
+* Configurable timeout for the rate limit service call in the :ref:`network
+  <config_network_filters_rate_limit>` and :ref:`HTTP <config_http_filters_rate_limit>` rate limit
+  filters.
+* :ref:`x-envoy-grpc-retry-on <config_http_filters_router_x-envoy-grpc-retry-on>` header added.
+* :ref:`LDS API <arch_overview_dynamic_config_lds>` added.
+* TLS :ref:`require_client_certificate <config_listener_ssl_context_require_client_certificate>`
+  option added.
+* Configuration check tool added.
+* Route table :ref:`check tool <config_tools_router_check_tool>` added.
+* JSON schema check tool added.
+* Config validation mode added via the :option:`--mode` option.
+* :option:`--local-address-ip-version` option added.
+* IPv6 support is now complete.
+* UDP :ref:`statsd_ip_address <config_overview_statsd_udp_ip_address>` option added.
+* Per-cluster :ref:`DNS resolvers <config_cluster_manager_cluster_dns_resolvers>` added.
+* :ref:`Fault filter <config_http_filters_fault_injection>` enhancements and fixes.
+* Several features are :repo:`deprecated as of the 1.4.0 release </DEPRECATED.md#version-140>`. They
+  will be removed at the beginning of the 1.5.0 release cycle.
```

Comment:
I think it's worth explicitly calling out the changes to the filter interface (`Named{Network,Http}FilterFactory`) and the removal of filter direction. In general we don't need fine grained deprecation in release notes, but these are major changes to how people write filters.